pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/generator/TaskGeneratorUtils.java

```diff
@@ -68,4 +73,30 @@
     }
     return runningSegments;
   }
+
+  public static Map<String, List<String>> getScheduledSegmentsMap(@Nonnull String taskType,
+      @Nonnull ClusterInfoProvider clusterInfoProvider) {
+    Map<String, List<String>> scheduledSegments = new HashMap<>();
+    Map<String, TaskState> taskStates = clusterInfoProvider.getTaskStates(taskType);
+    for (Map.Entry<String, TaskState> entry : taskStates.entrySet()) {
+      // Skip COMPLETED tasks
+      if (entry.getValue() == TaskState.COMPLETED) {
+        continue;
+      }
```

Comment:
Skip tasks that are older than one day?

Reply:
If the task got stuck, we are cleaning up the segment lineage data after one day (the retention manager runs the clean up for segment lineage).

Once the segment lineage is updated, the segments from the old task will be rescheduled. If we reschedule the segment here before the segment lineage gets cleaned up, the minion task will anyway fail because it won't be able to update segment lineage because we don't allow the same segment appear on the `segmentsFrom` field twice.

Reply:
Ah that makes sense. I added the part to skip tasks that are older than one day.